src/contracts/erc721.cairo

```diff
@@ -0,0 +1,297 @@
+use starknet::ContractAddress;
+
+#[starknet::interface]
+pub trait IERC721<TContractState> {
```

Comment:
- The reason for using a custom ERC721 contract is because of the functionality required of the mint function, which openzeppelin ERC721 does not have:
<img width="824" alt="image" src="https://github.com/user-attachments/assets/1f234cb0-3fd3-43e8-bbd4-5bb0aa6d1a33">

- The issue also stated that any ERC721 contract can be used: 
<img width="504" alt="image" src="https://github.com/user-attachments/assets/493dac65-7a02-43b8-aa9a-9a849e047624">

@tekkac